crates/storage/src/ops/mmr.rs

```diff
@@ -0,0 +1,15 @@
+//! MMR database operation interface.
+
+use strata_db_types::traits::MmrDatabase;
+use strata_merkle::MerkleProofB32 as MerkleProof;
+
+use crate::exec::*;
+
+inst_ops_simple! {
+    (<D: MmrDatabase> => MmrDataOps) {
+        append_leaf(hash: [u8; 32]) => u64;
+        generate_proof(index: u64) => MerkleProof;
+        generate_proofs(start: u64, end: u64) => Vec<MerkleProof>;
```

Comment:
I think this should still probably be being done at a higher layer instead of making the database understand how to do it.  Since then it has to be aware of the hash function we're using for the mmr.

Also, which state index is the proof valid against?  What if we have to generate one for a slightly older state?